diff -up pegasus/src/Pegasus/Common/SSLContext.cpp.openssl pegasus/src/Pegasus/Common/SSLContext.cpp
--- pegasus/src/Pegasus/Common/SSLContext.cpp.openssl	2008-12-16 19:56:00.000000000 +0100
+++ pegasus/src/Pegasus/Common/SSLContext.cpp	2009-08-26 14:22:18.000000000 +0200
@@ -272,7 +272,7 @@ int SSLCallback::verificationCRLCallback
     X509_REVOKED* revokedCert = NULL;
     for (int i = 0; i < sk_X509_REVOKED_num(revokedCerts); i++)
     {
-        revokedCert = (X509_REVOKED *)sk_value(X509_CRL_get_REVOKED(crl), i);
+        revokedCert = sk_X509_REVOKED_value(X509_CRL_get_REVOKED(crl), i);
 
         //a matching serial number indicates revocation
         if (ASN1_INTEGER_cmp(revokedCert->serialNumber, serialNumber) == 0)
